File: packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/hub/FlutterHubEvent.kt
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.amazonaws.amplify.amplify_datastore.types.hub

interface FlutterHubEvent {
    val eventName: String
    fun toValueMap(): Map<String, Any> {
        return mapOf(
            "eventName" to this.eventName
        )
    }
}

File: packages/amplify_datastore/darwin/amplify_datastore/Sources/amplify_datastore/FlutterHubEvent.swift
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import Foundation

protocol FlutterHubEvent {
    var eventName: String { get set }
    func toValueMap() -> [String: Any]
}

func shortEventName(eventName: String) -> String {
    return eventName.replacingOccurrences(of: "DataStore.", with: "")
}
